saw-remote-api/scripts/check_docs.sh

```diff
@@ -0,0 +1,18 @@
+#! /bin/bash
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+echo "Checking saw-remote-api docs (SAW.rst) are up-to-date with server"
+
+cd $DIR/../docs
+
+export SAW_SERVER=$(cabal v2-exec which saw-remote-api)
+if [[ -x "$SAW_SERVER" ]]; then
+  echo "using saw-remote-api at $SAW_SERVER"
+else
+  echo "cabal could not locate saw-remote-api via the which command"
+  exit 1
+fi
+
+$SAW_SERVER doc > TEMP.rst
+diff SAW.rst TEMP.rst
```

Comment:
I'm not sure how exactly this works. As far as I understand it, `saw-remote-api doc` just lists the documentation for the top-level commands, while [`SAW.rst`](https://github.com/GaloisInc/saw-script/blob/a9e13c2dc94725c69aec04a9b800a09684e5b51c/saw-remote-api/docs/SAW.rst) goes into quite a bit more depth than that. At the very least, a raw `diff` invocation seems likely to fail.

Reply:
Interesting. So it looks like `SAW.rst` was where we were growing/maintaining hand written documentation for the SAW server. Later, the `doc` server command was [added](https://github.com/GaloisInc/argo/commit/dc500fe3576481dc48b443c7e5420f2046e23abc) to Argo as a way to have the Haskell code take over responsibility for generating documentation (since it can use the type system to enforce that server methods and their parameters have at least _some_ documentation) and this feature eventually made its way here to the `saw-remote-api` [via a submodule bump PR with tweaks to get things working](https://github.com/GaloisInc/saw-script/commit/6d6b6a213a9ccf23213c8d0c37398c4fb4a910dc#diff-637c8f2f1bd5591f33d2aebdbd203cdd2e91b5e4c6e856ccde7c356d8c5d3470), but we (I) failed to fully unify our manual and automated documentation efforts.

I've added back the original manual docs as `old-Saw.rst` and left `SAW.rst` as the auto-generated version which gets checked in CI. I think we just need to make an issue to port the relevant added details from the manual docs over to the Haskell source code so it is automatically generated by `doc` as well.

Reply:
Relevant issue https://github.com/GaloisInc/saw-script/issues/1121